sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -430,7 +417,10 @@ public final class StorageBlobClient: PipelineClient {
         guard let url = self.url(forTemplate: urlTemplate, withKwargs: pathParams) else { return }
 
         // Construct query
-        var queryParams: [QueryParameter] = []
+        let queryParams = QueryParameters([
+            ("snapshot", options?.snapshot),
```

Comment:
Feels like the right place to convert this to a RFC1123 format string is here rather than making an alias and adding protocol conformance. But if you still want to do that without our special QueryStringConvertible protocol, you could do it by defining the alias, and then conforming to CustomStringConvertible instead.

Reply:
Doing the conversion here means that Autorest.Swift, its stencil files and its backing code models need to be increasingly more complicated and hard to maintain in order to know which string to render. That is why the purpose of this PR is to ensure that AzureCore handles all complexity so that all lines can be resolved to the following pattern: `("{key}": {path}{value})`